### PR TITLE
Filter product totals

### PR DESCRIPTION
**Pull request by wilco375**
This PR adds a filter field to the product totals so you can filter the product totals on a specific user or pin or cash transactions. Also see #379 

Default:
image

Filter on user:
image

Filter on pin transactions:
image

Autocomplete:
image

### CODE REVIEW

File: app/javascript/components/activity/producttotals.vue [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        if (this.user.id > 0) {
+          url += '?user='+this.user.id;
+        }
+        if (this.user.paid_with_cash) {
+          url += '?paid_with_cash=true';
+        }
+        if (this.user.paid_with_pin) {
+          url += '?paid_with_pin=true';
+        }
+        this.$http.get(url).then((response) => {
```
Comment (Matthijsy) [inline app/javascript/components/activity/producttotals.vue]: can't you pass an params argument to the get method? Then you could do `.get(url, params=this.user)`

File: app/javascript/components/userinput.vue [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      name: {
+        type: String
+      },
+      value: {
+        type: Object,
+        default: {
+          id: 0
+        }
+      },
+      includePayment: {
```
Comment (Matthijsy) [inline app/javascript/components/userinput.vue]: Might be better to call it something like `inlucudeNonUserPayment` (maybe bit long) or `includePin` and `includeCash`, with users is it also payment, so payment is a bit vague

Comment (wilco375) [reply app/javascript/components/userinput.vue]: Added includePin/includeCash

File: app/javascript/components/activity/producttotals.vue [hunk 3]
```diff
@@ -56,17 +56,17 @@
       loadProductTotals() {
         this.isLoading = true;
 
-        let url = '/activities/'+this.activity+'/product_totals';
+        let params = {};
```
Comment (Matthijsy) [inline app/javascript/components/activity/producttotals.vue]: What about:
```
let params = {user: this.user.id, paid_with_cash: this.user.paid_with_cash, paid_with_pin: this.user.paid_with_pin }
```

Does that work?

Comment (wilco375) [reply app/javascript/components/activity/producttotals.vue]: Yes!

**Comment by Matthijsy** [comment]
@wilco375 I've added the necessary specs

**Pull request merged**